fuse_models/include/fuse_models/odometry_2d_publisher.h

```diff
@@ -200,6 +201,11 @@ class Odometry2DPublisher : public fuse_core::AsyncPublisher
   std::unique_ptr<tf2_ros::TransformListener> tf_listener_;
 
   ros::Timer publish_timer_;
+
+  /**
+   * @brief A ros::console filter to print delayed throttle messages, that can be reset
+   */
+  std::unique_ptr<ros::console::FilterBase> delayed_throttle_filter_{ new fuse_core::DelayedThrottleFilter(10.0) };
```

Comment:
I have to ask if this really needs to be base-class pointer. It seems to me this could be:
```
fuse_core::DelayedThrottleFilter delayed_throttle_filter_ { 10.0 };
```
And then the reset operation becomes simply
```
delayed_throttle_filter_.reset();
```
And the console call becomes
```
ROS_WARN_STREAM_FILTER(&delayed_throttle_filter_, "No valid state data yet. Delaying tf broadcast.");
```

I did a quick test and that compiles, which is admittedly different from "it works".

Reply:
Oh, IIRC it failed to compile for me in `ROS_WARN_STREAM_FILTER(&delayed_throttle_filter_`, but it actually works now for me.

Changed.